src/lib/factory.js

```diff
@@ -178,16 +181,21 @@ class PluginFactory extends EventEmitter2 {
 
     // stop plugin from double-connecting
     plugin.disconnect = function () { return Promise.resolve(null) }
-    plugin.connect = function () { return Promise.resolve(null) }
+    plugin.connect = function () {
+      // TODO this should wait until the admin is subscribed to the account notifications
```

Comment:
If `connect` is changed to only resolve once it's subscribed to accounts, we don't have to worry about this case